server.js

```diff
@@ -0,0 +1,386 @@
+      if (
+        !formDataObject["elementName"] ||
+        !formDataObject["elementSymbol"] ||
+        !formDataObject["elementAtomicNumber"] ||
+        !formDataObject["elementDescription"]
+      ) {
+        console.log("Request is missing one or more values");
+        request.on("end", function() {
+          response.writeHead(500, { "Content-Type": "application/json" });
+          let responseBody = {
+            error: "Requeste is missing one or more values"
+          };
+          response.end(JSON.stringify(responseBody));
+        });
+        // Return 500 if file name is index or 404
+      } else if (
+        formDataObject["elementName"] === "index" ||
+        formDataObject["elementName"] === "404"
+      ) {
+        console.log("Invalid elementName");
+        request.on("end", function() {
+          response.writeHead(500, { "Content-Type": "application/json" });
+          let responseBody = {
+            error: "Invalid elementName"
+          };
+          response.end(JSON.stringify(responseBody));
+        });
+        // Create new HTML file and populate with POST data
+        let newElement = `<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <title>The Elements - ${formDataObject["elementName"]}</title>
+    <link rel="stylesheet" href="/css/styles.css" />
+  </head>
+  <body>
+    <h1>${formDataObject["elementName"]}</h1>
+    <h2>${formDataObject["elementSymbol"]}</h2>
+    <h3>Atomic number ${formDataObject["elementAtomicNumber"]}</h3>
+    <p>
+      ${formDataObject["elementDescription"]}
+    </p>
+    <p><a href="/">back</a></p>
+  </body>
+</html>
+`;
+        fs.writeFile(
+          "./public/" + formDataObject["elementName"].toLowerCase() + ".html",
+          newElement,
+          function(error) {
+            if (error) throw error;
+            console.log("File created successfully");
+          }
+        );
+        // Update index.html
+        fs.readFile("./public/index.html", function(error, data) {
+          if (error) throw error;
+
+          // Add element to end of list
+          let indexHTML = data.toString();
+          if (indexHTML.search(formDataObject["elementName"]) === -1) {
+            let addedElement = indexHTML.replace(
+              "</ol>",
+              `  <li>
+        <a href="/${formDataObject["elementName"].toLowerCase()}.html">${
+                formDataObject["elementName"]
+              }</a>
+      </li>
+    </ol>`
+            );
+            // Update the count of elements
+            let split = addedElement.split("\n");
+            split.splice(10, 1);
+            let join = split.join("\n").replace(
+              "</h2>",
+              `</h2>
+    <h3>There are ${addedElement.split("<li>").length - 1}</h3>`
+            );
+            // Write to index.html
+            fs.writeFile("./public/index.html", join, function(error) {
+              if (error) throw error;
```

Comment:
Errors should be handled properly, they should send back a response with a appropriate status code and message.